### PR TITLE
Fixed Grease Pencil nodes for latest Blender API changes.

### PR DESCRIPTION
**Pull request by 3DSinghVFX**
I have fixed the Grease Pencil nodes for the latest [Blender 2.83 API](https://wiki.blender.org/wiki/Reference/Release_Notes/2.83/Grease_Pencil) changes.
Changes are:
1. Mainly related to the layer mask.
2. Small bug fixes.
3. Added _Hardeness_ attribute to GP Stroke.

### CODE REVIEW

**Comment by 3DSinghVFX** [comment]
@OmarEmaraDev I have added the _Vertex Color_ attribute to the _GP Stroke_ object.

We can add _Vertex Color_ input in these nodes which would be very handy,
Screenshot from 2020-03-29 23-51-51

What do you think?

**Comment by OmarEmaraDev** [comment]
@3DSinghVFX I don't see how vertex colors could be incorporated in those nodes, aside from the offset node. Can you elaborate?

**Comment by 3DSinghVFX** [comment]
> @3DSinghVFX I don't see how vertex colors could be incorporated in those nodes, aside from the offset node. Can you elaborate?

If we integrate the _Vertex Color_ in _Offset GP Stroke_  node then it is not needed for remaining nodes (especially the _GP Stroke From Spline_ and _Transform GP Stroke_ nodes). However, we can integrate _Vertex Color_ in _Replicate GP Stroke_ node, it will allow the replication of stroke with different vertex colors.

**Comment by OmarEmaraDev** [comment]
I don't think we should add that to the Replicate node. The Offset node would be fine I guess.

**Comment by 3DSinghVFX** [comment]
> I don't think we should add that to the Replicate node. The Offset node would be fine I guess.

Okay. I'll make changes.

**Comment by 3DSinghVFX** [comment]
@OmarEmaraDev I have added the _offset color_ method with three basic color methods _ADD, MULTIPLY, MIX_ in `color/c_utils` file, and added the _Vertex Color_ offset option in the _Offset GP Stroke_ node.

**Comment by 3DSinghVFX** [comment]
Okay. I have done testing on the released Blender 2.83 build, it is working fine.

To get multiple layers by their name, we have to use multiple _GP Object Input_ nodes. Should we add _Get GP Layer From List_ node that will give a layer by name from the layer-list?
Screenshot from 2020-06-04 11-36-00

File: animation_nodes/nodes/gpencil/gp_layer_from_frames.py [hunk 1]
```diff
@@ -1,36 +1,71 @@
...
+                if self.isEmptyLayer(maskLayer): continue
+                maskLayerName = maskLayer.layerName
+                if maskLayerName != "" and maskLayerName != layerName:
+                    maskLayer.invertAsMask = invertMaskLayers[i]
+                    maskLayers.append(maskLayer)
+
+        return GPLayer(layerName, frames, blendMode, opacity, useLights, tintColor, tintFactor, lineChange,
+                       passIndex, False, maskLayers)
+
+    def isEmptyLayer(self, layer):
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/gpencil/gp_layer_from_frames.py:63]: You can move this to the GPLayer class as a class method, notice that it is also used in `set_gp_layer_attributes.py` Also, you can do something like this:
```
def isEmptyLayer(self):
    for frame in self.frames:
        if len(frame.strokes) > 0:
            return False
    return True
```

File: animation_nodes/nodes/gpencil/gp_stroke_from_points.py [hunk 2]
```diff
@@ -1,39 +1,44 @@
...
-    def execute(self, vertices, strengths, pressures, uvRotations, lineWidth, drawCyclic,
-                startCapMode, endCapMode, materialIndex, displayMode):
+    def execute(self, vertices, strengths, pressures, uvRotations, vertexColors, lineWidth,
+                hardness, drawCyclic, startCapMode, endCapMode, materialIndex, displayMode):
 
         amount = len(vertices)
         strengths = FloatList.fromValues(VirtualDoubleList.create(strengths, 1).materialize(amount))
         pressures = FloatList.fromValues(VirtualDoubleList.create(pressures, 1).materialize(amount))
         uvRotations = FloatList.fromValues(VirtualDoubleList.create(uvRotations, 0).materialize(amount))
+        vertexColors = ColorList.fromValues(VirtualColorList.create(vertexColors, Color((0, 0, 0, 0))).materialize(amount))
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/gpencil/gp_stroke_from_points.py:41]: The output of `materialize` is already a ColorList, so no need for `ColorList.fromValues` here.

File: animation_nodes/nodes/gpencil/gp_layer_from_frames.py (hunk 1 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/gp_layer_from_frames.py:63]: Okay.

File: animation_nodes/nodes/gpencil/gp_stroke_from_points.py (hunk 2 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/gp_stroke_from_points.py:41]: Okay.

File: animation_nodes/nodes/gpencil/gp_layer_from_frames.py (hunk 1 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/gp_layer_from_frames.py]: Done. BTW why class method not as regular method?

File: animation_nodes/nodes/gpencil/gp_stroke_from_points.py (hunk 2 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/gp_stroke_from_points.py]: Done.

**Comment by 3DSinghVFX** [comment]
> @3DSinghVFX Not in this patch. Lets talk about this later.

Okay.

File: animation_nodes/nodes/gpencil/gp_layer_from_frames.py (hunk 1 above)
Comment (OmarEmaraDev) [reply animation_nodes/nodes/gpencil/gp_layer_from_frames.py]: @3DSinghVFX My apologies, I meant a normal method.

Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/gp_layer_from_frames.py]: Okay, I'll change that.

Should I removed this `if len(maskLayersIn) > 0`?. I added for the performance:
https://github.com/JacquesLucke/animation_nodes/blob/5c8005435cd226c5f3783899b7137bd0f4795fcf/animation_nodes/nodes/gpencil/gp_layer_from_frames.py#L50-L58

Comment (OmarEmaraDev) [reply animation_nodes/nodes/gpencil/gp_layer_from_frames.py]: @3DSinghVFX  It shouldn't make much of a difference. It is up to you.

Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/gp_layer_from_frames.py]: Okay. I'll keep that.

**Pull request merged**